plugins/scanners/fossid/src/test/kotlin/events/UploadArchiveHandlerTest.kt

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2025 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.scanners.fossid.events
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.file.shouldExist
+import io.kotest.matchers.file.shouldNotExist
+
+import io.mockk.mockk
+
+import java.io.File
+
+import kotlin.io.path.createTempDirectory
+
+import org.ossreviewtoolkit.model.config.Excludes
+import org.ossreviewtoolkit.model.config.Includes
+import org.ossreviewtoolkit.model.config.PathExclude
+import org.ossreviewtoolkit.model.config.PathExcludeReason
+import org.ossreviewtoolkit.model.config.config.PathInclude
+import org.ossreviewtoolkit.model.config.config.PathIncludeReason
+import org.ossreviewtoolkit.plugins.scanners.fossid.createConfig
+import org.ossreviewtoolkit.plugins.scanners.fossid.createServiceMock
+import org.ossreviewtoolkit.scanner.provenance.NestedProvenance
+
+class UploadArchiveHandlerTest : WordSpec({
+    "deleteExcludedFiles()" should {
+        "delete files that match exclusion patterns" {
+            val tempDir = createTempDirectory("test-delete-excluded").toFile()
```

Comment:
Generally, use Kotest's `tempdir()` / `tempfile()` in tests as these get deleted automatically.